st2common/st2common/util/api.py

```diff
@@ -140,3 +141,17 @@ def get_exception_for_type_error(func, exc):
         result = exc
 
     return result
+
+
+def get_exception_for_uncaught_api_error(func, exc):
+    """
+    Utility function which tries to map an uncaught exception throwed inside an
+    API to a more user-friendly exception which is returned instead of returning
+    internal server error.
+    """
+
+    if isinstance(exc, mongoengine.ValidationError):
+        result = webob_exc.HTTPBadRequest(detail=exc.message)
+        return result
```

Comment:
Should have been a part of https://github.com/StackStorm/st2/blob/master/st2common/st2common/hooks.py#L245. Took me quite a while to figure out the hell is going on.